File: non_parallel.py
import os
import numpy as np
from concurrent.futures import ProcessPoolExecutor

DIR = "simulation_results"

def simulate_and_save(m, paramter_one, paramter_two):
    # Simulate data and run model

    # Here, you would replace this with actual simulation code
    # For demonstration, we will just add two parameters and some random noise
    np.random.seed(m)
    result = paramter_one + paramter_two + np.random.normal(0, 1, 1)
    
    # Save results to csv
    np.savetxt(f"{DIR}/results_{m}.csv", result, delimiter=",")
    return f"Simulation {m} saved."

def main():
    paramter_one = 1
    paramter_two = 2
    
    # Number of simulations
    num_simulations = 64
    
    # Ensure the directory exists
    if not os.path.exists(DIR):
        os.makedirs(DIR, exist_ok=True)

    for m in range(num_simulations):
        simulate_and_save(m, paramter_one, paramter_two)

if __name__ == "__main__":
    main()

File: R/non_parallel.R

DIR <- "simulation_results"

simulate_and_save <- function(m, parameter_one, parameter_two) {
  set.seed(m)
  result <- parameter_one + parameter_two + rnorm(1, 0, 1)
  
  # Save results to csv
  write.csv(result, file = paste0(DIR, "/results_", m, ".csv"), row.names = FALSE)
  return(paste("Simulation", m, "saved."))
}

main <- function() {
  parameter_one <- 1
  parameter_two <- 2
  
  # Number of simulations
  num_simulations <- 64
  
  # Ensure the directory exists before the loop
  if(!dir.exists(DIR)) {
    dir.create(DIR, recursive = TRUE)
  }
  
  # Run simulations sequentially using a for loop
  results <- vector("list", num_simulations) # Pre-allocate storage for results
  for (m in 1:num_simulations) {
    results[[m]] <- simulate_and_save(m, parameter_one, parameter_two)
  }
  
  print(results)
}

# Run the main function
main()
